atspi-proxies/src/collection.rs

```diff
@@ -1,60 +1,95 @@
-	  &[i32]: roles,
-	  i32: AtspiCollectionMatchType (role match type),
-	  &[&str]: interfaces,
-	  i32: AtspiCollectionMatchType (interface match type),
-	  bool: invert
-	*/
-	/// GetMatches method
+	/// Retrieves a list of objects that match the specified ObjectMatchRule, ordered according to SortOrder and limited by the count parameter.
+	///
+	/// # Arguments
+	///
+	/// * `rule` - An [`ObjectMatchRule`] describing the match criteria.
+	/// * `sortby` - A [`SortOrder`] specifying the way the results are to be sorted.
+	/// * `count` - The maximum number of results to return, or 0 for no limit.
+	/// * `traverse` - Not supported.
+	///
+	/// [`ObjectMatchRule`]: atspi_common::object_match::ObjectMatchRule
+	/// [`SortOrder`]: atspi_common::SortOrder
 	fn get_matches(
 		&self,
-		rule: &MatchArgs<'_>,
+		rule: ObjectMatchRule,
 		sortby: SortOrder,
 		count: i32,
 		traverse: bool,
 	) -> zbus::Result<Vec<ObjectRef>>;
 
-	/// GetMatchesFrom method
+	/// Retrieves objects from the collection, after `current_object`, matching a given `rule`.
+	///
+	/// # Arguments
+	///
+	/// * `current_object` - The object at which to start searching.
+	/// * `rule` - An [`ObjectMatchRule`] describing the match criteria.
+	/// * `sortby` - A [`SortOrder`] specifying the way the results are to be sorted.
+	/// * `tree` - A [`TreeTraversalType`] specifying restrictions on the objects to be traversed.
+	/// * `limit_scope` - If `true`, only descendants of `current_object`'s parent will be returned. Otherwise (if `false`), any accessible may be returned if it would preceed `current_object` in a flattened hierarchy.
+	/// * `count` - The maximum number of results to return, or 0 for no limit.
+	/// * `traverse` - Not supported.
+	///
+	/// [`ObjectMatchRule`]: atspi_common::object_match::ObjectMatchRule
+	/// [`SortOrder`]: atspi_common::SortOrder
+	/// [`TreeTraversalType`]: atspi_common::TreeTraversalType
 	fn get_matches_from(
 		&self,
 		current_object: &zbus::zvariant::ObjectPath<'_>,
-		rule: &MatchArgs<'_>,
+		rule: ObjectMatchRule,
 		sortby: SortOrder,
 		tree: TreeTraversalType,
 		count: i32,
 		traverse: bool,
 	) -> zbus::Result<Vec<ObjectRef>>;
 
-	/// GetMatchesTo method
+	/// Retrieves objects from the collection, before `current_object`, matching a given `rule`.
+	///
+	/// # Arguments
+	///
+	/// * `current_object` - The object at which to start searching.
+	/// * `rule` - An [`ObjectMatchRule`] describing the match criteria.
+	/// * `sortby` - A [`SortOrder`] specifying the way the results are to be sorted.
+	/// * `tree` - A [`TreeTraversalType`] specifying restrictions on the objects to be traversed.
+	/// # Maximum number of objects
+	///
+	/// This method will never return more than 65536 objects.
+	///
+	/// [`ObjectMatchRule`]: atspi_common::object_match::ObjectMatchRule
+	/// [`SortOrder`]: atspi_common::SortOrder
+	/// [`TreeTraversalType`]: atspi_common::TreeTraversalType
+	#[allow(clippy::too_many_arguments)]
```

Comment:
There is a `limit_scope` here, so perhaps these functions got their docs reversed?

Reply:
Yes, they look alike.